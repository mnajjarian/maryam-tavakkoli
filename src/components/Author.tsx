import React, { useContext } from "react";
import { DataContext } from "../contexts/dataContext";

interface Props {
  name: String;
}
const Author = (props: Props) => {
  const {
    data: { profile }
  } = useContext(DataContext);
  if (!profile) {
    return <div></div>;
  }
  const author = profile.filter((p: any) => p.name === props.name)[0];

  return (
    <>
      <h2>About the author</h2>
      <img
        className="author__img"
        src={require(`../assets/images/bio-image.jpg`)}
        alt="author"
      />
      <p className="author__name">
        <strong>{author.name}</strong>
      </p>
      <p></p>
      <p className="author__bio">{author.biography}</p>
    </>
  );
};

export default Author;
