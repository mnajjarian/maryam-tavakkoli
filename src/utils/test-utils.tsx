import React from "react";
import { render } from "@testing-library/react";
import { AuthProvider } from "../contexts/authContext";
import { DataProvider } from "../contexts/dataContext";
import { ReactNode } from "react";
import { BrowserRouter, Router } from "react-router-dom";
import { createBrowserHistory } from "history";

const history = createBrowserHistory();
interface Props {
  children: ReactNode;
}
const AllTheProviders = (props: Props) => {
  return (
    <AuthProvider>
      <DataProvider>
        <BrowserRouter>
          <Router history={history}>{props.children}</Router>
        </BrowserRouter>
      </DataProvider>
    </AuthProvider>
  );
};

const customRender = (ui: any, options: any) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
