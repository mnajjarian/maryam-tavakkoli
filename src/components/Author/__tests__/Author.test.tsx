import React from "react";
import Author from "..";
import { render } from "test-utils";

test("renders Author without crashing ", async () => {
  const component = render(<Author />);
  expect(component.container).toMatchSnapshot();
});
