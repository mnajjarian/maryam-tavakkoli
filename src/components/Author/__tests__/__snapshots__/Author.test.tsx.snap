// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Author without crashing  1`] = `
<div>
  <div />
</div>
`;
